servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -439,16 +366,14 @@ public void completeFile(long fileId)
       InodeFile fileInode = (InodeFile) inode;
       List<Long> blockIdList = fileInode.getBlockIds();
       List<BlockInfo> blockInfoList = mBlockMaster.getBlockInfoList(blockIdList);
-      if (blockInfoList.size() != blockIdList.size()) {
```

Comment:
This old check doesn't make sense if the file is persisted. However, doesn't this check still make sense when the file is NOT persisted? When the file is not persisted, then this check would make sure all the blocks in the file are committed in Tachyon space. Otherwise, iterating over all the blocks below could have missing blocks.